scripts/prove_rpc.rs

```diff
@@ -0,0 +1,196 @@
+use std::{
+    env::set_var,
+    fmt::Display,
+    fs::create_dir_all,
+    path::{Path, PathBuf},
+};
+
+use alloy::{eips::BlockId, transports::http::reqwest::Url};
+use anyhow::Result;
+use clap::{arg, Args, ValueEnum, ValueHint};
+
+use crate::process::Process;
+
+#[derive(ValueEnum, Clone)]
+enum RpcType {
+    Jerigon,
+    Native,
+}
+
+impl Display for RpcType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            RpcType::Jerigon => write!(f, "jerigon"),
+            RpcType::Native => write!(f, "native"),
+        }
+    }
+}
+
+#[derive(ValueEnum, Copy, Clone)]
+enum RunMode {
+    /// Dummy proof is generated. Useful for quickly testing decoding and
+    /// all other non-proving logic.
+    Test,
+    /// The proof is generated but is not verified.
+    Prove,
+    /// The proof is generated and verified.
+    Verify,
+}
+
+#[derive(Args)]
+pub struct ProveRpcArgs {
+    /// The node RPC URL.
+    #[arg(value_hint = ValueHint::Url)]
+    rpc_url: Url,
+    /// The RPC type (jerigon or native).
+    #[arg()]
+    rpc_type: RpcType,
+    /// Whether to generate a proof and verify it or not.
+    #[arg()]
```

Comment:
nit: you don't need `arg` here
```suggestion
    rpc_type: RpcType,
    /// Whether to generate a proof and verify it or not.
```